modules/ROOT/pages/dataweave-cookbook-extract-data.adoc

```diff
@@ -4,287 +4,200 @@ include::_attributes.adoc[]
-is `application/json`.
-
-For more examples, see <<key_value>>.
-
-| *Descendants* (`..`)
-a|
-`..` selector acts on arrays and objects.
-
-* Selecting values from the `c` key:
-+
-```
-{ "user" : [{ "a" : "b"}, {"c" : [ {"aa" : "bb"}, { "cc": "dd" } ] } ] , "user" : "d" }..c
-```
-+
-Returns `[ [ { "aa": "bb" }, { "cc": "dd" } ] ]`.
-
-For more examples, see <<descendants>>.
-
-| *Index* (`[]`)
-a|
-`[]` selector acts on indices of an array.
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+var myObject = { "myKey" : "1234", "name" : "somebody" }
+var myArray = [ { "myKey" : "1234" }, { "name" : "somebody" } ]
+output application/json
+---
+{
+    selectingValueUsingKeyInObject : myObject.name,
+    selectingValueUsingKeyOfObjectInArray : myArray.name,
+}
+----
 
-* Selecting an index from an array:
-** `[1,2,3,4][0]` returns `[1]`
-+
-The index of the first element in an array is always `0`.
-+
-** `[1,2,3,4][4]` returns `[3]`
+.Output:
+[source,JSON,linenums]
+----
+{
+    "selectingValueUsingKeyInObject": "somebody",
+    "selectingValueUsingKeyOfObjectInArray": [ "somebody" ]
+ }
+----
 
-For more examples, see <<range>>.
+As the Output shows:
 
-| *Range* `[index1 to index2]`
-a| `[index1 to index2]` selector produces and array with values from selected
-indexes.
+* `myObject.name` returns the value `"somebody"`
+* `myArray.name` returns the array `[ "somebody" ]`
 
-* Selecting a range of indices:
-** `[0,1,2][0 to 1]` returns `[ 0, 1 ]`
-** `[0,1,2][-1 to 0]` returns an inverted list: `[ 2, 1, 0 ]`
+== Selector Quick Reference
 
-For more examples, see xref:dataweave-selectors#range-selector.adoc[Range Selectors]
+Single Value (`.`)::
+Acts on arrays and objects to return the value of a matching key. The syntax
+is `.myKey`. To retrieve values of duplicate keys in a DataWeave _object_, use
+`*`, instead. For examples, see <<single_value>>.
 
-| *XML attribute* `.@keyName`
-a| `.@keyName` selector returns the value of the selected attribute.
+Multiple Values (`*`)::
+Acts on arrays and objects to retrieve the values of all matching keys
+at a single level in the hierarchy of a data structure. The syntax is
+`*myKey`. For values of all duplicate keys at lower levels in the hierarchy,
+use the descendants selector (`..`), instead. For examples, see <<multi_value>>.
 
-For an example, see <<xml_attribute>>.
+Key-Value Pair (`&amp;`)::
+Acts on arrays and objects. Instead of returning the value of the DataWeave
+object, this selector returns the entire DataWeave object, both key and value.
+The syntax is `.&amp;myKey`. For examples, see <<key_value>>.
 
-| *Namespace Selector* `keyName.#`
-a| `keyName.#` selector returns the namespace of the selected XML element.
+Descendants (`..`)::
+Acts on arrays and objects to retrieve all matching keys from arrays and
+object below the given key, regardless of their location in the hierarchy.
+The syntax is `..myKey` For more examples, see <<descendants>>.
```

Comment:
"For examples" (to be consistent with the other entries)

Reply:
done